internal/controller/instaslice_controller.go

```diff
@@ -548,42 +613,72 @@ func (l *RightToLeftPolicy) SetAllocationDetails(profileName string, newStart, s
 	return &inferencev1alpha1.AllocationDetails{}
 }
 
-func (r *InstasliceReconciler) processInstasliceAllocation(ctx context.Context, instasliceName string, podUUID string, allocation inferencev1alpha1.AllocationDetails, req ctrl.Request) (bool, ctrl.Result, error) {
-	if allocation.PodUUID == podUUID {
-		if allocation.Allocationstatus == inferencev1alpha1.AllocationStatusDeleted {
-			deleteResult, errDeletingAllocation := r.deleteInstasliceAllocation(ctx, instasliceName, allocation)
-			if errDeletingAllocation != nil {
-				return true, deleteResult, errDeletingAllocation
-			}
+func (r *InstasliceReconciler) removeInstasliceAllocation(ctx context.Context, instasliceName string, allocation inferencev1alpha1.AllocationDetails) (ctrl.Result, error) {
+	if allocation.Allocationstatus == inferencev1alpha1.AllocationStatusDeleted {
+		deleteResult, errDeletingAllocation := r.deleteInstasliceAllocation(ctx, instasliceName, allocation)
```

Comment:
Could be simplified to just return r.deleteInstasliceAllocation(...

Reply:
same as https://github.com/openshift/instaslice-operator/pull/141#discussion_r1793904839